### PR TITLE
chore: adds Python 3.7/3.8 EOL pending deprecation warning

### PR DESCRIPTION
**Pull request by chalmerlowe**
This is a work in progress (WIP). I want to see how it behaves in our CI/CD pipeline and if that is any different than what we see locally. 

Adds a pending deprecation warning and code logic to output that warning if the user attempts to install the repo using Python 3.7 or 3.8.

The expected outcome is that the system will display a `PendingDeprecationWarning` if it detects Python runtimes 3.7 or 3.8, will display nothing if using more recent Python runtimes 3.9 to 3.x

### CODE REVIEW

**Comment by chalmerlowe** [comment]
Before approving, we are looking for a blog post URL to add.
I don't yet know what the URL will be.

File: google/cloud/bigquery/__init__.py [hunk 1]
```diff
@@ -114,12 +114,25 @@
...
 
+sys_major, sys_minor, sys_micro = _versions_helpers.extract_runtime_version()
+
+if sys_major == 3 and sys_minor in (7, 8):
+    warnings.warn(
+        "The python-bigquery library will stop supporting Python 3.7 "
+        "and Python 3.8 in a future major release expected in Q4 2024. "
+        f"Your Python version is {sys_major}.{sys_minor}.{sys_micro}. We "
+        "recommend that you update soon to ensure ongoing support. For "
+        "more details, see: <blog post URL>",
```
Comment (chalmerlowe) [inline google/cloud/bigquery/__init__.py:132]:
```suggestion
        "more details, see: [Google Cloud Client Libraries Supported Python Versions policy](https://cloud.google.com/python/docs/supported-python-versions)",
```

**Comment by chalmerlowe** [comment]
@Linchin 

I updated the URL with one that points at the Google policy.

Can you please review this?

Thanks.

**Review by Linchin — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**